File: MoveZeroes.cs
// c# solution for move zeroes

public class Solution {
    public void MoveZeroes(int[] nums) {
        var pointer1 = 0;
        
        for (int i = 0; i < nums.Length; i++)
            if (nums[i] != 0)
                nums[pointer1++] = nums[i];
        
        for (int j = pointer1; j < nums.Length; j++)
            nums[j] = 0;
    }
}

File: moveZeroes.js
// Given an array nums, write a function to move all 0's to the end of it while maintaining the relative order of the non-zero elements.

// Example:

// Input: [0,1,0,3,12]
// Output: [1,3,12,0,0]

var moveZeroes = function(nums) {
    let zeroCount = 0;
       for (let i = 0; i < nums.length; i++) {
           if (nums[i] === 0) {
               nums.splice(i,1)
               zeroCount++
               i--
           }
       }
   
       for (let i = 0; i < zeroCount; i++) {
         nums.push(0)
       }
   };